engine/polyglot-api/src/main/scala/org/enso/polyglot/runtime/Runtime.scala

```diff
@@ -108,20 +108,18 @@ object Runtime {
     /** An update about the computed expression.
       *
       * @param expressionId the expression id
-      * @param expressionType the type of expression
+      * @param expressionTypes the type of expression
       * @param methodCall the underlying method call of this expression
-      * @param profilingInfo profiling information about the execution of this
-      * expression
-      * @param fromCache whether or not the value for this expression came
-      * from the cache
-      * @param typeChanged whether or not the type of the value or method definition
+      * @param profilingInfo profiling information about the execution of this expression
+      * @param fromCache whether the value for this expression came from the cache
+      * @param typeChanged whether the type of the value or method definition
       * has changed from the one that was cached, if any
       * @param payload an extra information about the computed value
       */
     @named("expressionUpdate")
     case class ExpressionUpdate(
       expressionId: ExpressionId,
-      expressionType: Option[String],
+      expressionTypes: Option[Vector[String]],
```

Comment:
Isn't that the point of this change? 
```
expression-types: null
```
(when `None`)
and
```
expression-types: []
```
are both allowed in JSON and yet represent two different scenarios and should be handled separately?
But maybe I'm missing something, @4e6 ?

Reply:
Jaroslav's point was that an empty vector is the same as `None`. But because it is a valid state in the engine, I made it this way to transfer a raw engine response to the language server. In the user-facing update message the type is `type: string[]`

Reply:
I see. I still have problems to properly identify the _"JSON protocol level"_ each class represents. I am overwhelmed by _too many JSON messages_ situation.